src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -317,11 +498,36 @@ - (void)startNextRead {
 
 #pragma mark Send headers
 
-- (void)sendHeaders:(NSDictionary *)headers {
+- (void)sendHeaders {
+  // TODO (mxyan): Remove after deprecated methods are removed
+  uint32_t callSafetyFlags;
+  switch (_callSafety) {
+    case GRPCCallSafetyDefault:
+      callSafetyFlags = 0;
+      break;
+    case GRPCCallSafetyIdempotentRequest:
+      callSafetyFlags = GRPC_INITIAL_METADATA_IDEMPOTENT_REQUEST;
+      break;
+    case GRPCCallSafetyCacheableRequest:
+      callSafetyFlags = GRPC_INITIAL_METADATA_CACHEABLE_REQUEST;
+      break;
+    default:
```

Comment:
By adding a default to the switch you're preventing clang from warning you about incomplete case coverage at compile time. You still need the runtime check, but consider putting the exception as a separate check.

Reply:
Didn't really understand this comment. I think the `default` case should not be happening so an exception seems to be reasonable here?